src/Catalyst.Core.Modules.Dfs.Tests/IntegrationTests/CoreApi/DagApiTest.cs

```diff
@@ -0,0 +1,156 @@
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Text;
+using System.Threading.Tasks;
+using Catalyst.Abstractions.Dfs;
+using Catalyst.Core.Modules.Dfs.Tests.Utils;
+using Newtonsoft.Json.Linq;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Catalyst.Core.Modules.Dfs.Tests.IntegrationTests.CoreApi
+{
+    public sealed class DagApiTest
+    {
+        private readonly byte[] _blob = Encoding.UTF8.GetBytes("blorb");
+        private const string Blob64 = "YmxvcmI"; // base 64 encoded with no padding
+        private readonly IDfsService _dfs;
+
+        public DagApiTest(ITestOutputHelper output)
+        {
+            _dfs = TestDfs.GetTestDfs(output, null, "sha2-256");
+        }
+
+        [Fact]
+        public async Task Get_Raw()
+        {
+            var cid = await _dfs.BlockApi.PutAsync(_blob, contentType: "raw");
+            Assert.Equal("bafkreiaxnnnb7qz2focittuqq3ya25q7rcv3bqynnczfzako47346wosmu", cid);
+
+            var dag = await _dfs.DagApi.GetAsync(cid);
+            Assert.Equal(Blob64, (string) dag["data"]);
+        }
+
+        private sealed class Name
+        {
+            public string First { get; set; }
+            public string Last { get; set; }
+        }
+
+        [Fact]
+        public async Task PutAndGet_JSON()
+        {
+            var expected = new JObject {["a"] = "alpha"};
+            const string expectedId = "bafyreigdhej736dobd6z3jt2vxsxvbwrwgyts7e7wms6yrr46rp72uh5bu";
+            var id = await _dfs.DagApi.PutAsync(expected);
+            Assert.NotNull(id);
+            Assert.Equal(expectedId, id);
+
+            var actual = await _dfs.DagApi.GetAsync(id);
+            Assert.NotNull(actual);
+            Assert.Equal(expected["a"], actual["a"]);
+
+            var value = (string) await _dfs.DagApi.GetAsync(expectedId + "/a");
+            Assert.Equal(expected["a"], value);
+        }
+
+        [Fact]
+        public async Task PutAndGet_poco()
+        {
+            var expected = new Name {First = "John", Last = "Smith"};
+            var id = await _dfs.DagApi.PutAsync(expected);
+            Assert.NotNull(id);
+
+            var actual = await _dfs.DagApi.GetAsync<Name>(id);
+            Assert.NotNull(actual);
+            Assert.Equal(expected.First, actual.First);
+            Assert.Equal(expected.Last, actual.Last);
+
+            var value = (string) await _dfs.DagApi.GetAsync(id.Encode() + "/Last");
+            Assert.Equal(expected.Last, value);
+        }
+
+        /// <summary>
+        /// @TODO hardcoded encoding types
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Complete the task associated to this 'TODO' comment.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)